File: EmployeeManagement/Models/Employee.cs
namespace EmployeeManagement.Models;

public class Employee
{
    public Guid Id { get; set; }

    public Guid UserId { get; set; }

    public string FirstName { get; set; } = null!;
    public string LastName { get; set; } = null!;

    public string Department { get; set; } = null!;

    public string Position { get; set; } = null!;

    public DateTime JoinedAt { get; set; }

    public ApplicationUser User { get; set; } = null!;
}

File: employee-management-client/src/types/employee.ts
export interface Employee {
    id: string;
    employeeId: string;
    userId: string;
    firstName: string;
    lastName: string;
    email: string;
    phoneNumber?: string;
    joinedAt: string;
    roles: string[];
    position:string;
    department:string;
    isDisabled:boolean;
}

export interface EmployeeCreateRequest {
    firstName: string;
    lastName: string;
    email: string;
    password: string;
    department: string;
    position: string;
}

export interface EmployeeUpdateRequest {
    firstName: string;
    lastName: string;
    email: string;
    department: string;
    position: string;
}

export interface UpdateRolesRequest {
    roles: string[];
}

export interface PaginationQuery {
    search?: string;
    page: number;
    pageSize: number;
}

export interface PagedResult<T> {
    items: T[];
    page: number;
    pageSize: number;
    totalCount: number;
}